lib/project_types/extension/commands/serve.rb

```diff
@@ -35,17 +24,43 @@ def self.help
 
       private
 
-      def argo_admin?
-        ShopifyCli::Shopifolk.check &&
-          ShopifyCli::Feature.enabled?(:argo_admin_beta) &&
-          specification_handler.specification.features&.argo&.surface == "admin"
-      end
+      def validate_env!
+        return unless ShopifyCli::Shopifolk.check &&
+          specification.feature?(:argo) &&
```

Comment:
This still feels fairly `argo` specific. Do you plan to push this logic into the `features` or `specification_handlers`?